### PR TITLE
nexus command to copy repo & sign

### PR DESCRIPTION
**Pull request by pkriens**

### CODE REVIEW

File: biz.aQute.repository/src/aQute/bnd/repository/maven/provider/Crawler.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+			uris.addAll(p.getValue());
+		}
+		return uris;
+	}
+
+	public List<URI> getURIs(URI uri) throws Exception {
+		return getURIs(uri, u -> true);
+	}
+
+	public static Predicate<URI> predicate(String include, String exclude) {
```
Comment (bjhargrave) [inline biz.aQute.repository/src/aQute/bnd/repository/maven/provider/Crawler.java]: This should use the new `PathSet.find()` methods to build the predicate from a set of includes and excludes. I just added these in https://github.com/bndtools/bnd/pull/3740/commits/5897aaf42ae697b01f5adea3f966d158ca726c38#diff-e731aefa8d049ad50ec427de5c0f25e7R175-R207 so you will need to rebase on master.

Changing to use PathSet, means you wont need to change to add Glob.NONE.

File: biz.aQute.repository/src/aQute/maven/nexus/provider/Nexus.java [hunk 2]
```diff
@@ -122,13 +122,16 @@ private boolean isReal(URI uri) {
...
+			.timeout(5 * 60 * 1000)
+			.retries(3)
 			.go(uri);
 	}
 
 	public void upload(URI uri, byte[] data) throws Exception {
 		try (TaggedData tag = request().put()
 			.upload(data)
 			.asTag()
+			.retries(3)
```
Comment (bjhargrave) [inline biz.aQute.repository/src/aQute/maven/nexus/provider/Nexus.java]: Currently retries are limited to `GET`. So we would need to change HttpClient to support them for other http verbs.

https://github.com/bndtools/bnd/blob/6ca9a2214a422eaa93fae07ec34aa6624885d385/biz.aQute.bndlib/src/aQute/bnd/http/HttpClient.java#L153

File: .gitignore [hunk 3]
```diff
@@ -8,4 +8,5 @@ Archive
 /reports/
 /generated/
 /install-jdk.sh
-.to
\ No newline at end of file
+/bndtools.test/
```
[the remark this answers is not in the document]
Comment (pkriens) [reply .gitignore]: ok, I do not have it for some reason.

File: biz.aQute.repository/src/aQute/bnd/repository/maven/provider/Crawler.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+					continue; // not recursive, points outside
+
+				Promise<List<URI>> downstream = pf.submit(() -> getURIs(next, predicate));
+				promises.add(downstream);
+			} else {
+				if (predicate.test(next))
+					uris.add(next);
+			}
+		}
+		for (Promise<List<URI>> p : promises) {
```
[the remark this answers is not in the document]
Comment (pkriens) [reply biz.aQute.repository/src/aQute/bnd/repository/maven/provider/Crawler.java]: I want the exception separate. And the method is synchronous, so at less code than handling the aggregate

File: biz.aQute.repository/src/aQute/bnd/repository/maven/provider/Crawler.java (hunk 1 above)
Comment (pkriens) [reply biz.aQute.repository/src/aQute/bnd/repository/maven/provider/Crawler.java]: Not completely clear how it works, you can fix that if you want.

**Pull request closed without merging**